event-driven/src/lib.rs

```diff
@@ -23,54 +23,43 @@ pub use event_driven_macros::impl_fsm;
 /// and then causing a state transition in relation to that result. While this approach adds
 /// steps to a state machine, it does allow them to remain responsive to receiving more
 /// commands.
-///
-/// The generic types refer to:
-/// S  = State          - the state of your FSM
-/// C  = Command        - the command(s) that are able to be processed on your FSM
-/// E  = Event          - the event(s) that are emitted having performed a command
-/// SE = State Effect   - the effect handler
-pub trait Fsm<S, C, E, SE> {
+pub trait Fsm: Sized {
```

Comment:
You never explained why Fsm requires Sized.  Is it still required now that Self is a unit type?

Reply:
Ah, no longer required. Good catch. Removed.